Jaeger exporter drops batches of spans exceeding the UDP packet size limit

### Description

The Jaeger exporter, when configured with and agent endpoint that it will spans to via UDP, will fail to send any spans if the batch of spans when serialized in the thrift buffer exceeds the UDP max packet size. Instead it will return [an error](https://github.com/open-telemetry/opentelemetry-go/blob/70bc9eb391965ba22e4d9f938dc300f7e9fbc128/exporters/trace/jaeger/agent.go#L119-L122) saying the data does not fit within one UDP packet and will drop the batch of spans.

### Environment

- OS: Linux 5.11.7-arch1-1
- Architecture: x86_64
- Go Version: 1.16
- opentelemetry-go version: [a2bf04d](https://github.com/open-telemetry/opentelemetry-go/commit/a2bf04dc367dbd2828423e629975935377ea7a65)

### Steps To Reproduce

1. Using:
   ```
   import (
   	"context"
   	"testing"
   
   	"github.com/stretchr/testify/assert"
   	"go.opentelemetry.io/otel"
   	"go.opentelemetry.io/otel/exporters/trace/jaeger"
   	tracesdk "go.opentelemetry.io/otel/sdk/export/trace"
   )
   
   type errorHandler struct{ t *testing.T }
   
   func (eh errorHandler) Handle(err error) { assert.NoError(eh.t, err) }
   
   func TestJaegerAgentUDPLimitBatching(t *testing.T) {
   	otel.SetErrorHandler(errorHandler{t})
   
   	n := 1500
   	s := make([]*tracesdk.SpanSnapshot, n)
   	for i := 0; i < n; i++ {
   		s[i] = &tracesdk.SpanSnapshot{}
   	}
   	exp, err := jaeger.NewRawExporter(
   		jaeger.WithAgentEndpoint("localhost:0"),
   		jaeger.WithBatchMaxCount(n+1),
   		jaeger.WithBufferMaxCount(n+1),
   	)
   	assert.NoError(t, err)
   
   	ctx := context.Background()
   	assert.NoError(t, exp.ExportSpans(ctx, s))
   	assert.NoError(t, exp.Shutdown(ctx))
   }
   ```
3. Run `go test .`
4. See
   ```
   --- FAIL: TestJaegerAgentUDPLimitBatching (0.00s)
       bug_test.go:15: 
           	Error Trace:	bug_test.go:15
           	            				handler.go:60
           	            				handler.go:88
           	            				jaeger.go:124
           	            				bundler.go:324
           	            				asm_amd64.s:1371
           	Error:      	Received unexpected error:
           	            	data does not fit within one UDP packet; size 79559, max 65000, spans 1500
           	Test:       	TestJaegerAgentUDPLimitBatching
   FAIL
   FAIL	testing/jaeger-udp-bug	0.007s
   FAIL
   ```

### Expected behavior

The exporter should batch any bundles larger than the max payload size. This is what [the Jaeger Go client does](https://github.com/jaegertracing/jaeger-client-go/blob/ba87d35db7991cab81ba9bed697d17a2433b83eb/transport_udp.go). Possibly that library should be leveraged for our transport or we need to implement similar logic to check the thrift buffer is appropriately split into send-able payloads.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -93,7 +93,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
   - `OTEL_EXPORTER_OTLP_TRACES_CERTIFICATE`
   - `OTEL_EXPORTER_OTLP_METRICS_CERTIFICATE`
 - Adds `otlpgrpc.WithTimeout` option for configuring timeout to the otlp/gRPC exporter. (#1821)
-- Adds `jaeger.WithMaxPacketSize` option for configuring maximum packet size of jaeger udp agent.
+- Adds `jaeger.WithMaxPacketSize` option for configuring maximum UDP packet size used when connecting to the Jaeger agent. (#1853)
 
 ### Fixed
 
@@ -105,7 +105,8 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 - Zipkin Exporter: Ensure mapping between OTel and Zipkin span data complies with the specification. (#1688)
 - Fixed typo for default service name in Jaeger Exporter. (#1797)
 - Fix flaky OTLP for the reconnnection of the client connection. (#1527, #1814)
-- Fix Jaeger exporter drops batches of spans exceeding the UDP packet size limit. (#1828)
+- Fix Jaeger exporter dropping of span batches that exceed the UDP packet size limit.
+  Instead, the exporter now splits the batch into smaller sendable batches. (#1828)
 
 ### Changed
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -93,6 +93,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
   - `OTEL_EXPORTER_OTLP_TRACES_CERTIFICATE`
   - `OTEL_EXPORTER_OTLP_METRICS_CERTIFICATE`
 - Adds `otlpgrpc.WithTimeout` option for configuring timeout to the otlp/gRPC exporter. (#1821)
+- Adds `jaeger.WithMaxPacketSize` option for configuring maximum packet size of jaeger udp agent.
 
 ### Fixed
 
@@ -104,6 +105,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 - Zipkin Exporter: Ensure mapping between OTel and Zipkin span data complies with the specification. (#1688)
 - Fixed typo for default service name in Jaeger Exporter. (#1797)
 - Fix flaky OTLP for the reconnnection of the client connection. (#1527, #1814)
+- Fix Jaeger exporter drops batches of spans exceeding the UDP packet size limit. (#1828)
 
 ### Changed
 

--- a/exporters/trace/jaeger/agent.go
+++ b/exporters/trace/jaeger/agent.go
@@ -36,10 +36,11 @@ type agentClientUDP struct {
 	genAgent.Agent
 	io.Closer
 
-	connUDP       udpConn
-	client        *genAgent.AgentClient
-	maxPacketSize int                   // max size of datagram in bytes
-	thriftBuffer  *thrift.TMemoryBuffer // buffer used to calculate byte size of a span
+	connUDP        udpConn
+	client         *genAgent.AgentClient
+	maxPacketSize  int                   // max size of datagram in bytes
+	thriftBuffer   *thrift.TMemoryBuffer // buffer used to calculate byte size of a span
+	thriftProtocol thrift.TProtocol
 }
 
 type udpConn interface {
@@ -75,6 +76,7 @@ func newAgentClientUDP(params agentClientUDPParams) (*agentClientUDP, error) {
 
 	thriftBuffer := thrift.NewTMemoryBufferLen(params.MaxPacketSize)
 	protocolFactory := thrift.NewTCompactProtocolFactoryConf(&thrift.TConfiguration{})
+	thriftProtocol := protocolFactory.GetProtocol(thriftBuffer)
 	client := genAgent.NewAgentClientFactory(thriftBuffer, protocolFactory)
 
 	var connUDP udpConn
@@ -103,15 +105,21 @@ func newAgentClientUDP(params agentClientUDPParams) (*agentClientUDP, error) {
 	}
 
 	return &agentClientUDP{
-		connUDP:       connUDP,
-		client:        client,
-		maxPacketSize: params.MaxPacketSize,
-		thriftBuffer:  thriftBuffer,
+		connUDP:        connUDP,
+		client:         client,
+		maxPacketSize:  params.MaxPacketSize,
+		thriftBuffer:   thriftBuffer,
+		thriftProtocol: thriftProtocol,
 	}, nil
 }
 
 // EmitBatch implements EmitBatch() of Agent interface
 func (a *agentClientUDP) EmitBatch(ctx context.Context, batch *gen.Batch) error {
+	return a.splitAndFlush(ctx, batch)
+}
+
+// Flush will send the batched spans though underlying thrift transport
+func (a *agentClientUDP) Flush(ctx context.Context, batch *gen.Batch) error {
 	a.thriftBuffer.Reset()
 	if err := a.client.EmitBatch(ctx, batch); err != nil {
 		return err
@@ -121,6 +129,46 @@ func (a *agentClientUDP) EmitBatch(ctx context.Context, batch *gen.Batch) error 
 			a.thriftBuffer.Len(), a.maxPacketSize, len(batch.Spans))
 	}
 	_, err := a.connUDP.Write(a.thriftBuffer.Bytes())
+	return err
+}
+
+// calcSizeOfSerializedThrift calculate the serialized thrift packet size
+func (a *agentClientUDP) calcSizeOfSerializedThrift(ctx context.Context, thriftStruct thrift.TStruct) int {
+	a.thriftBuffer.Reset()
+	_ = thriftStruct.Write(ctx, a.thriftProtocol)
+	return a.thriftBuffer.Len()
+}
+
+// splitAndFlush split large batch into small ones to fit the udp packet limit
+func (a *agentClientUDP) splitAndFlush(ctx context.Context, batch *gen.Batch) error {
+	var err error
+	processSize := a.calcSizeOfSerializedThrift(ctx, batch.Process)
+	totalSize := processSize
+	var spans []*gen.Span
+	for _, span := range batch.Spans {
+		spanSize := a.calcSizeOfSerializedThrift(ctx, span)
+		if spanSize+processSize >= a.maxPacketSize {
+			// drop the span that exceeds the limit
+			continue
+		}
+		if totalSize+spanSize >= a.maxPacketSize {
+			err = a.Flush(ctx, &gen.Batch{
+				Process: batch.Process,
+				Spans:   spans,
+			})
+			spans = spans[:0]
+			totalSize = processSize
+		}
+		totalSize += spanSize
+		spans = append(spans, span)
+	}
+
+	if len(spans) > 0 {
+		err = a.Flush(ctx, &gen.Batch{
+			Process: batch.Process,
+			Spans:   spans,
+		})
+	}
 	return err
 }
 

--- a/exporters/trace/jaeger/agent.go
+++ b/exporters/trace/jaeger/agent.go
@@ -115,11 +115,56 @@ func newAgentClientUDP(params agentClientUDPParams) (*agentClientUDP, error) {
 
 // EmitBatch implements EmitBatch() of Agent interface
 func (a *agentClientUDP) EmitBatch(ctx context.Context, batch *gen.Batch) error {
-	return a.splitAndFlush(ctx, batch)
+	var errs []error
+	processSize, err := a.calcSizeOfSerializedThrift(ctx, batch.Process)
+	if err != nil {
+		// drop the bath if serialization of process fails.
+		return err
+	}
+	totalSize := processSize
+	var spans []*gen.Span
+	for _, span := range batch.Spans {
+		spanSize, err := a.calcSizeOfSerializedThrift(ctx, span)
+		if err != nil {
+			errs = append(errs, fmt.Errorf("thrift serialization failed: %v", span))
+			continue
+		}
+		if spanSize+processSize >= a.maxPacketSize {
+			// drop the span that exceeds the limit.
+			errs = append(errs, fmt.Errorf("span too large to send: %v", span))
+			continue
+		}
+		if totalSize+spanSize >= a.maxPacketSize {
+			if err := a.flush(ctx, &gen.Batch{
+				Process: batch.Process,
+				Spans:   spans,
+			}); err != nil {
+				errs = append(errs, err)
+			}
+			spans = spans[:0]
+			totalSize = processSize
+		}
+		totalSize += spanSize
+		spans = append(spans, span)
+	}
+
+	if len(spans) > 0 {
+		if err := a.flush(ctx, &gen.Batch{
+			Process: batch.Process,
+			Spans:   spans,
+		}); err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	if len(errs) > 0 {
+		return fmt.Errorf("multiple errors during transform")
+	}
+	return nil
 }
 
-// Flush will send the batched spans though underlying thrift transport
-func (a *agentClientUDP) Flush(ctx context.Context, batch *gen.Batch) error {
+// flush will send the batch of spans to the agent.
+func (a *agentClientUDP) flush(ctx context.Context, batch *gen.Batch) error {
 	a.thriftBuffer.Reset()
 	if err := a.client.EmitBatch(ctx, batch); err != nil {
 		return err
@@ -132,44 +177,11 @@ func (a *agentClientUDP) Flush(ctx context.Context, batch *gen.Batch) error {
 	return err
 }
 
-// calcSizeOfSerializedThrift calculate the serialized thrift packet size
-func (a *agentClientUDP) calcSizeOfSerializedThrift(ctx context.Context, thriftStruct thrift.TStruct) int {
+// calcSizeOfSerializedThrift calculate the serialized thrift packet size.
+func (a *agentClientUDP) calcSizeOfSerializedThrift(ctx context.Context, thriftStruct thrift.TStruct) (int, error) {
 	a.thriftBuffer.Reset()
-	_ = thriftStruct.Write(ctx, a.thriftProtocol)
-	return a.thriftBuffer.Len()
-}
-
-// splitAndFlush split large batch into small ones to fit the udp packet limit
-func (a *agentClientUDP) splitAndFlush(ctx context.Context, batch *gen.Batch) error {
-	var err error
-	processSize := a.calcSizeOfSerializedThrift(ctx, batch.Process)
-	totalSize := processSize
-	var spans []*gen.Span
-	for _, span := range batch.Spans {
-		spanSize := a.calcSizeOfSerializedThrift(ctx, span)
-		if spanSize+processSize >= a.maxPacketSize {
-			// drop the span that exceeds the limit
-			continue
-		}
-		if totalSize+spanSize >= a.maxPacketSize {
-			err = a.Flush(ctx, &gen.Batch{
-				Process: batch.Process,
-				Spans:   spans,
-			})
-			spans = spans[:0]
-			totalSize = processSize
-		}
-		totalSize += spanSize
-		spans = append(spans, span)
-	}
-
-	if len(spans) > 0 {
-		err = a.Flush(ctx, &gen.Batch{
-			Process: batch.Process,
-			Spans:   spans,
-		})
-	}
-	return err
+	err := thriftStruct.Write(ctx, a.thriftProtocol)
+	return a.thriftBuffer.Len(), err
 }
 
 // Close implements Close() of io.Closer and closes the underlying UDP connection.

--- a/exporters/trace/jaeger/agent.go
+++ b/exporters/trace/jaeger/agent.go
@@ -20,6 +20,7 @@ import (
 	"io"
 	"log"
 	"net"
+	"strings"
 	"time"
 
 	"go.opentelemetry.io/otel/exporters/trace/jaeger/internal/third_party/thrift/lib/go/thrift"
@@ -113,12 +114,12 @@ func newAgentClientUDP(params agentClientUDPParams) (*agentClientUDP, error) {
 	}, nil
 }
 
-// EmitBatch implements EmitBatch() of Agent interface
+// EmitBatch buffers batch to fit into UDP packets and sends the data to the agent.
 func (a *agentClientUDP) EmitBatch(ctx context.Context, batch *gen.Batch) error {
 	var errs []error
 	processSize, err := a.calcSizeOfSerializedThrift(ctx, batch.Process)
 	if err != nil {
-		// drop the bath if serialization of process fails.
+		// drop the batch if serialization of process fails.
 		return err
 	}
 	totalSize := processSize
@@ -157,10 +158,22 @@ func (a *agentClientUDP) EmitBatch(ctx context.Context, batch *gen.Batch) error 
 		}
 	}
 
-	if len(errs) > 0 {
-		return fmt.Errorf("multiple errors during transform")
+	if len(errs) == 1 {
+		return errs[0]
+	} else if len(errs) > 1 {
+		joined := a.makeJoinedErrorString(errs)
+		return fmt.Errorf("multiple errors during transform: %s", joined)
 	}
 	return nil
+}
+
+// makeJoinedErrorString join all the errors to one error message.
+func (a *agentClientUDP) makeJoinedErrorString(errs []error) string {
+	var errMsgs []string
+	for _, err := range errs {
+		errMsgs = append(errMsgs, err.Error())
+	}
+	return strings.Join(errMsgs, ", ")
 }
 
 // flush will send the batch of spans to the agent.

--- a/exporters/trace/jaeger/agent_test.go
+++ b/exporters/trace/jaeger/agent_test.go
@@ -21,6 +21,7 @@ import (
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+
 	"go.opentelemetry.io/otel"
 	tracesdk "go.opentelemetry.io/otel/sdk/trace"
 )
@@ -110,6 +111,7 @@ func (eh errorHandler) Handle(err error) { assert.NoError(eh.t, err) }
 func TestJaegerAgentUDPLimitBatching(t *testing.T) {
 	otel.SetErrorHandler(errorHandler{t})
 
+	// 1500 spans, size 79559, does not fit within one UDP packet with the default size of 65000.
 	n := 1500
 	s := make([]*tracesdk.SpanSnapshot, n)
 	for i := 0; i < n; i++ {
@@ -119,9 +121,31 @@ func TestJaegerAgentUDPLimitBatching(t *testing.T) {
 	exp, err := NewRawExporter(
 		WithAgentEndpoint(WithAgentHost("localhost"), WithAgentPort("6831")),
 	)
-	assert.NoError(t, err)
+	require.NoError(t, err)
 
 	ctx := context.Background()
 	assert.NoError(t, exp.ExportSpans(ctx, s))
+	assert.NoError(t, exp.Shutdown(ctx))
+}
+
+func TestSpanExceedsMaxPacketLimit(t *testing.T) {
+	otel.SetErrorHandler(errorHandler{t})
+
+	// 102 is the serialized size of a span with default values.
+	maxSize := 102
+	span := &tracesdk.SpanSnapshot{
+		Name: "a-longger-name-that-make-it-exceeds-limit",
+	}
+	largeSpans := []*tracesdk.SpanSnapshot{span, {}}
+	normalSpans := []*tracesdk.SpanSnapshot{{}, {}}
+
+	exp, err := NewRawExporter(
+		WithAgentEndpoint(WithAgentHost("localhost"), WithAgentPort("6831"), WithMaxPacketSize(maxSize+1)),
+	)
+	require.NoError(t, err)
+
+	ctx := context.Background()
+	assert.Error(t, exp.ExportSpans(ctx, largeSpans))
+	assert.NoError(t, exp.ExportSpans(ctx, normalSpans))
 	assert.NoError(t, exp.Shutdown(ctx))
 }

--- a/exporters/trace/jaeger/agent_test.go
+++ b/exporters/trace/jaeger/agent_test.go
@@ -14,12 +14,15 @@
 package jaeger
 
 import (
+	"context"
 	"log"
 	"net"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/otel"
+	tracesdk "go.opentelemetry.io/otel/sdk/trace"
 )
 
 func TestNewAgentClientUDPWithParamsBadHostport(t *testing.T) {
@@ -98,4 +101,27 @@ func TestNewAgentClientUDPWithParamsReconnectingDisabled(t *testing.T) {
 	assert.IsType(t, &net.UDPConn{}, agentClient.connUDP)
 
 	assert.NoError(t, agentClient.Close())
+}
+
+type errorHandler struct{ t *testing.T }
+
+func (eh errorHandler) Handle(err error) { assert.NoError(eh.t, err) }
+
+func TestJaegerAgentUDPLimitBatching(t *testing.T) {
+	otel.SetErrorHandler(errorHandler{t})
+
+	n := 1500
+	s := make([]*tracesdk.SpanSnapshot, n)
+	for i := 0; i < n; i++ {
+		s[i] = &tracesdk.SpanSnapshot{}
+	}
+
+	exp, err := NewRawExporter(
+		WithAgentEndpoint(WithAgentHost("localhost"), WithAgentPort("6831")),
+	)
+	assert.NoError(t, err)
+
+	ctx := context.Background()
+	assert.NoError(t, exp.ExportSpans(ctx, s))
+	assert.NoError(t, exp.Shutdown(ctx))
 }

--- a/exporters/trace/jaeger/agent_test.go
+++ b/exporters/trace/jaeger/agent_test.go
@@ -131,12 +131,13 @@ func TestJaegerAgentUDPLimitBatching(t *testing.T) {
 func TestSpanExceedsMaxPacketLimit(t *testing.T) {
 	otel.SetErrorHandler(errorHandler{t})
 
-	// 102 is the serialized size of a span with default values.
-	maxSize := 102
+	// 106 is the serialized size of a span with default values.
+	maxSize := 106
 	span := &tracesdk.SpanSnapshot{
-		Name: "a-longger-name-that-make-it-exceeds-limit",
+		Name: "a-longer-name-that-makes-it-exceeds-limit",
 	}
-	largeSpans := []*tracesdk.SpanSnapshot{span, {}}
+	oneLargeSpan := []*tracesdk.SpanSnapshot{span}
+	largeSpans := []*tracesdk.SpanSnapshot{span, span, {}}
 	normalSpans := []*tracesdk.SpanSnapshot{{}, {}}
 
 	exp, err := NewRawExporter(
@@ -145,6 +146,7 @@ func TestSpanExceedsMaxPacketLimit(t *testing.T) {
 	require.NoError(t, err)
 
 	ctx := context.Background()
+	assert.Error(t, exp.ExportSpans(ctx, oneLargeSpan))
 	assert.Error(t, exp.ExportSpans(ctx, largeSpans))
 	assert.NoError(t, exp.ExportSpans(ctx, normalSpans))
 	assert.NoError(t, exp.Shutdown(ctx))

--- a/exporters/trace/jaeger/agent_test.go
+++ b/exporters/trace/jaeger/agent_test.go
@@ -128,16 +128,22 @@ func TestJaegerAgentUDPLimitBatching(t *testing.T) {
 	assert.NoError(t, exp.Shutdown(ctx))
 }
 
+// generateALargeSpan generates a span with a long name.
+func generateALargeSpan() *tracesdk.SpanSnapshot {
+	span := &tracesdk.SpanSnapshot{
+		Name: "a-longer-name-that-makes-it-exceeds-limit",
+	}
+	return span
+}
+
 func TestSpanExceedsMaxPacketLimit(t *testing.T) {
 	otel.SetErrorHandler(errorHandler{t})
 
 	// 106 is the serialized size of a span with default values.
 	maxSize := 106
-	span := &tracesdk.SpanSnapshot{
-		Name: "a-longer-name-that-makes-it-exceeds-limit",
-	}
-	oneLargeSpan := []*tracesdk.SpanSnapshot{span}
-	largeSpans := []*tracesdk.SpanSnapshot{span, span, {}}
+	span := generateALargeSpan()
+
+	largeSpans := []*tracesdk.SpanSnapshot{span, {}}
 	normalSpans := []*tracesdk.SpanSnapshot{{}, {}}
 
 	exp, err := NewRawExporter(
@@ -146,8 +152,25 @@ func TestSpanExceedsMaxPacketLimit(t *testing.T) {
 	require.NoError(t, err)
 
 	ctx := context.Background()
-	assert.Error(t, exp.ExportSpans(ctx, oneLargeSpan))
 	assert.Error(t, exp.ExportSpans(ctx, largeSpans))
 	assert.NoError(t, exp.ExportSpans(ctx, normalSpans))
 	assert.NoError(t, exp.Shutdown(ctx))
+}
+
+func TestEmitBatchWithMultipleErrors(t *testing.T) {
+	otel.SetErrorHandler(errorHandler{t})
+
+	span := generateALargeSpan()
+	largeSpans := []*tracesdk.SpanSnapshot{span, span}
+	// make max packet size smaller than span
+	maxSize := len(span.Name)
+	exp, err := NewRawExporter(
+		WithAgentEndpoint(WithAgentHost("localhost"), WithAgentPort("6831"), WithMaxPacketSize(maxSize)),
+	)
+	require.NoError(t, err)
+
+	ctx := context.Background()
+	err = exp.ExportSpans(ctx, largeSpans)
+	assert.Error(t, err)
+	require.Contains(t, err.Error(), "multiple errors")
 }

--- a/exporters/trace/jaeger/uploader.go
+++ b/exporters/trace/jaeger/uploader.go
@@ -114,7 +114,7 @@ func WithAttemptReconnectingInterval(interval time.Duration) AgentEndpointOption
 	}
 }
 
-// WithMaxPacketSize sets the maximum size of udp packet.
+// WithMaxPacketSize sets the maximum UDP packet size for transport to the Jaeger agent.
 func WithMaxPacketSize(size int) AgentEndpointOption {
 	return func(o *AgentEndpointOptions) {
 		o.MaxPacketSize = size

--- a/exporters/trace/jaeger/uploader.go
+++ b/exporters/trace/jaeger/uploader.go
@@ -114,6 +114,13 @@ func WithAttemptReconnectingInterval(interval time.Duration) AgentEndpointOption
 	}
 }
 
+// WithMaxPacketSize sets the maximum size of udp packet.
+func WithMaxPacketSize(size int) AgentEndpointOption {
+	return func(o *AgentEndpointOptions) {
+		o.MaxPacketSize = size
+	}
+}
+
 // WithCollectorEndpoint defines the full url to the Jaeger HTTP Thrift collector. This will
 // use the following environment variables for configuration if no explicit option is provided:
 //

